pkg/api/types.go

```diff
@@ -2134,6 +2167,84 @@ func (cs *ContainerService) GetAzureProdFQDN() string {
 	return FormatProdFQDNByLocation(cs.Properties.MasterProfile.DNSPrefix, cs.Location, cs.Properties.GetCustomCloudName())
 }
 
+// ProvisionScriptParametersInput is the struct used to pass in Azure environment variables and secrets
+// as either values or ARM template variables when generating provision script parameters.
+type ProvisionScriptParametersInput struct {
+	Location             string
+	ResourceGroup        string
+	TenantID             string
+	SubscriptionID       string
+	ClientID             string
+	ClientSecret         string
+	APIServerCertificate string
+	KubeletPrivateKey    string
+	ClusterKeyVaultName  string
+}
+
+// GetProvisionScriptParametersCommon returns the environment variables needed to run the Linux bootstrap scripts
+// Ensure that the clientSecret parameter is surrounded by single quotes to protect against special characters
+func (cs *ContainerService) GetProvisionScriptParametersCommon(input ProvisionScriptParametersInput) string {
+	cloudSpecConfig := cs.GetCloudSpecConfig()
+	kubernetesConfig := cs.Properties.OrchestratorProfile.KubernetesConfig
+	provisionScriptParametersCommonString := "" +
```

Comment:
nit:
You could use strings.Builder

```
var b strings.Builder
b.WriteString("...")
b.WriteString("...")
return b.String()
```

Reply:
Yeah that's a good point. It does make the concatenation more efficient but will make the code harder to read because it will require putting each WriteString on a new line (ie. double the lines it's on now) and the string for the env var name won't be next to its value anymore. Thoughts?

Reply:
I think it would allow you to break that method in a smaller one: 
`AddParameter(builder, name, value)`

Reply:
How about something like this?

https://play.golang.org/p/-oi0IU1_N36